<launch>

<!-- Default folder with maps-->
<arg name="folder" default="$(env HOME)/test_bearnav"/>

<!-- Distance Measurement Node -->
<node pkg="stroll_bearnav" type="odometry_monitor_node" name="odometry_monitor" output="screen">
    <remap from="odom" to="odometry/filtered"/>
    <remap from="joint_states" to="dev/null"/>
</node>

<!-- feature extraction -->
<node pkg="stroll_bearnav" type="feature_extraction_node" name="feature_extraction">
    <remap from="image" to="/ros_indigosdk_occam/image0"/>
    <param name="image_transport" value="compressed"/>
</node>

<!-- mapping node -->
<node pkg="stroll_bearnav" type="mapper_node" name="mapper" output="screen">
    <remap from="image" to="/ros_indigosdk_occam/image0"/>
    <param name="image_transport" value="compressed"/>
    <param name="folder" value="$(arg folder)" />
</node>

<!-- map loader node -->
<node pkg="stroll_bearnav" type="map_preprocessor_node" name="map_preprocessor" output="screen">
<param name="folder" value="$(arg folder)" />
</node>

<!-- navigator node -->
<node pkg="stroll_bearnav" type="navigator_node" name="navigator" output="screen">
    <param name="folder" value="$(arg folder)" />
    <remap from="cmd" to="jackal_velocity_controller/cmd_vel"/>
</node>

</launch>
